lib/injection.dart

```diff
@@ -0,0 +1,8 @@
+import 'package:get_it/get_it.dart';
+
+import 'data/dataBase/data_base_handler.dart';
+
+GetIt sl = GetIt.instance;
+Future initGetIt() async {
+  sl.registerFactory(() => DatabaseHandler());
```

Comment:
**registerFactory()**: receives a factory function that returns `a NEW instance` of implementation of T. Each time you call `get<T>()` you will get `a new instance returned`.

**BUT**:

**registerLazySingleton()**: receives a factory function that returns an instance of an implementation of T. Only the `first time` you call `get<T>()` this factory function will be called to create a `new instance`. After that, you will `always` get the `same instance returned`.

This means you should register `DB` as a `LazySingleton` instead of `Factory` because you don't want to build the database each time you use or call `get()`